# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/animation/Animation.cpp
# Commit: 63603267402e
# Full Hash: 63603267402ee93d6551ef60c35191d99aaee06b
# Author: Zach Hoffman <zach@zrhoffman.net>
# Date: 2023-12-18 17:17:57
# Regressor Bug: 1846516
# File Overlap Count: 3
# Description:
#   Bug 1846516 - [css-properties-values-api] Introduce AnimatedPropertyID/AnimatedPropertIDSet. r=firefox-animation-reviewers,firefox-style-system-reviewers,layout-reviewers,emilio,hiro
#   
#   This will make possible to animate custom properties on Gecko side. For now, the
#   animation code keeps only dealing with nsCSSPropertyID, so behavior is unchanged.
#   
# ==============================================================================

diff -r 753bc39f59ff -r 63603267402e dom/animation/Animation.cpp
--- a/dom/animation/Animation.cpp	Mon Dec 18 09:24:07 2023 +0000
+++ b/dom/animation/Animation.cpp	Mon Dec 18 09:24:08 2023 +0000
@@ -868,10 +868,11 @@
 
   // Set the animated styles
   bool changed = false;
-  nsCSSPropertyIDSet properties = keyframeEffect->GetPropertySet();
-  for (nsCSSPropertyID property : properties) {
+  const AnimatedPropertyIDSet& properties = keyframeEffect->GetPropertySet();
+  for (const AnimatedPropertyID& property : properties) {
+    // TODO(zrhoffman, bug 1846516): Handle custom properties
     RefPtr<StyleAnimationValue> computedValue =
-        Servo_AnimationValueMap_GetValue(animationValues.get(), property)
+        Servo_AnimationValueMap_GetValue(animationValues.get(), property.mID)
             .Consume();
     if (computedValue) {
       changed |= Servo_DeclarationBlock_SetPropertyToAnimationValue(